include/swift/Remote/MetadataReader.h

```diff
@@ -193,8 +193,18 @@ class MetadataReader {
   /// amounts of data when we encounter corrupt values for sizes/counts.
   static const uint64_t MaxMetadataSize = 1048576; // 1MB
 
-  /// A cache of built types, keyed by the address of the type.
-  std::unordered_map<StoredPointer, BuiltType> TypeCache;
+  /// Define a has function for a std::pair<StoredPointer, bool>
+  struct TypeCacheKeyHash {
+    std::size_t operator()(const std::pair<StoredPointer, bool> &x) const {
+      return std::hash<StoredPointer>()(x.first) ^ std::hash<bool>()(x.second);
```

Comment:
There's a hash_combine function in `stdlib/include/llvm/ADT/Hashing.h` that will do a little better job of combining these two.